cmd/boost/direct_deal.go

```diff
@@ -105,24 +258,49 @@ var directDealAllocate = &cli.Command{
 			return fmt.Errorf("failed to get allocations: %w", err)
 		}
 
-		mcid, sent, err := lib.SignAndPushToMpool(cctx, ctx, gapi, n, msg)
-		if err != nil {
-			return err
+		var mcids []cid.Cid
+
+		for _, msg := range msgs {
+			mcid, sent, err := lib.SignAndPushToMpool(cctx, ctx, gapi, n, msg)
+			if err != nil {
+				return err
```

Comment:
Message failure is a little tricky. One way would be to simply print the CIDs we already have sent successfully and users can get the param from those and decode into a list of piece details. Other way would be to print the allocation list and a diff can be used to remove the already published allocation details from the file and then user can retry.
I could print the success list in case of failure but that could be huge depending upon the input file and not a pretty output. 

Let me know your thoughts.

Reply:
does this relate to the CSV input thing so that the file drops out and the format could get re-absorbed with a manual run?

Reply:
I did not understand that comment. Can you clarify?

Reply:
Yes, error retries are difficult to handle. You can only succeed once, otherwise it will take a lot of time to filter out the wrong entries. This is the cost of using on-chain submissions.